protocols/luci-proto-bonding/htdocs/luci-static/resources/protocol/bonding.js

```diff
@@ -0,0 +1,442 @@
+'use strict';
+'require ui';
+'require uci';
+'require form';
+'require network';
+
+function getSelectableSlaves(section_id) {
+	var rv = [];
+	var interfaces = uci.sections('network', 'interface');
+
+	return network.getDevices().then(function(devices) {
+		for (var i = 0; i < devices.length; i++) {
+			var in_use = false;
+			if (devices[i].ifname.match(/eth/)) {
```

Comment:
Please use `devices[i]._devstate('type') == 1` to select ethernet interfaces only.

Reply:
If I use your suggested change, I get more than only the eth interfaces returned (e.g. erspan0, wwan0, wwan1 and -indeed- former created bonding interfaces). That is not what I want. I really want only the eth interfaces. So, unfortunately, your suggestion doesn't work for me. Any other suggestions?

Reply:
In addition to type `1`, try filtering for `devices[i]._devstate('flags', 'noarp') == false` too.

Reply:
Unfortunately Your suggestions didn't work as wanted, so I fixed with MAC address check.